outrank/algorithms/feature_ranking/ranking_mi_numba.py

```diff
@@ -82,11 +82,17 @@ def compute_entropies(
         initial_prob = _f_value_counts / all_events
         x_value_subspace = np.where(X == f_values[f_index])
 
-        Y_classes = Y[x_value_subspace]
-        Y_classes_spoofed = np.roll(Y, _f_value_counts)[x_value_subspace]
+        Y_classes = Y[x_value_subspace].astype(np.uint32)
+        subspace_size = x_value_subspace[0].size
 
-        nonzero_class_counts = np.zeros(len(class_values), dtype=np.int32)
-        nonzero_class_counts_spoofed = np.zeros(len(class_values), dtype=np.int32)
+        # Right-shift to simulate noise
+        Y_classes_spoofed = np.zeros(subspace_size, dtype=np.uint32)
+        for enx, el in enumerate(x_value_subspace[0]):
+            index = (el + _f_value_counts) % len(Y)
+            Y_classes_spoofed[enx] = Y[index]
```

Comment:
Hi @miha-jenko, you raise a very interesting question. Tested also this version, and it seems to yield similar results + behaves similarly w.r.t. control baselines (cardinality impact is mitigated indeed). However, if doing that, we're sampling from a potentially very different distribution -- for each value, random backdrop will be the same in this case. If there is some random inconsistency at the beginning of Y (there is no randomization), it could be this might skew the results. E.g., very sparse input vectors imply there could be no correction, even though for particular value this should be the case (didn't test this on extensive synthetic data of course). Finally, if you benchmark this, I actually see no speedup due to this simplification (that introduces another bias). Perhaps I'm missing something apparent though :smile_cat: